docs/source/traits_api_reference/trait_handlers.rst

```diff
@@ -27,15 +23,33 @@ Classes
 
 .. autoclass:: TraitCompound
 
-.. autoclass:: TraitTuple
-
-.. autoclass:: TraitList
-
-.. autoclass:: TraitDict
-
 Private Functions
 -----------------
 
 .. autofunction:: traits.trait_handlers._undefined_get
 
 .. autofunction:: traits.trait_handlers._undefined_set
+
+Deprecated Handlers
+-------------------
+
+.. deprecated:: 6.0.0
+
+The following :class:`~.TraitHandler` classes and instances are deprecated,
+and may be removed in a future version of Traits.
+
+.. autodata:: ThisClass
```

Comment:
I think we may be able to remove `ThisClass` altogether for the 6.0 release. There's little risk of breakage: it's (almost certainly) unused outside Traits, modulo some mentions in the TraitsUI docs. Those could be fixed.

Reply:
(TraitsUI doc fix: enthought/traitsui#724)